oauth2_provider/oauth2_backends.py

```diff
@@ -180,7 +180,11 @@ def extract_body(self, request):
         :return: provided POST parameters "urlencodable"
         """
         try:
-            body = json.loads(request.body.decode('utf-8')).items()
+            data = json.loads(request.body.decode('utf-8'))
+            if isinstance(data, dict):
+                body = data.items()
+            else:
+                body = ""
```

Comment:
You could also solve this by catching `AttributeError` on the except block and won't need to add any extra code

```
try:
    body = json.loads(request.body.decode("utf-8")).items()
except AttributeError, ValueError:
    body = ""
```

it's simpler, don't you think?